### PR TITLE
Use setuptools-scm for versioning

### PR DESCRIPTION
**Pull request by Nothing4You**
## What do these changes do?

Use setuptools-scm for versioning
- generate aiomysql/_scm_version.py at build time, fallback to `unknown` if it doesn't exist
- include setuptools-scm-git-archive to support versioning in archives from git tags
- install aiomysql package in readthedocs and access generated version to embed in docs

## To do
- [x] Update docs `Makefile` to fetch git tags to ensure consistent versioning for local builds, see https://github.com/aio-libs/aiomysql/pull/734#discussion_r812405601

## Are there changes in behavior for the user?

Self-reported `__version__` now returns version generated by `setuptools-scm` during build, otherwise `'unknown'`.

## Related issue number

split from #734

## Checklist

- [ ] I think the code is well written
- [ ] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [x] Add a new news fragment into `CHANGES.txt`

### CODE REVIEW

**Comment by Nothing4You** [comment]
readthedocs builds don't currently have enough git context to determine the correct version, see build output for the `pip install .` step in the build log: https://readthedocs.org/projects/aiomysql/builds/16470357/

this relates to https://github.com/readthedocs/readthedocs.org/issues/5031, I've requested the related feature flag via support form.

**Comment by Nothing4You** [comment]
RTD now fetches the full git history, showing correct scm generated version numbers.

**Comment by Nothing4You** [comment]
the only issue left with this is that `make doc` only works in an environment that has the package installed, other than that this should be good to go.

File: .git_archival.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+ref-names: $Format:%D$
```
Comment (webknjaz) [inline .git_archival.txt:1]: @Nothing4You FYI the recent versions of setuptools-scm need a more complete template — the current one has 4 lines, not just one.

**Pull request merged**